service/core/client/src/main/scala/com/lightbend/lagom/internal/client/WebSocketClient.scala

```diff
@@ -402,11 +399,13 @@ class WebSocketException(s: String, th: Throwable) extends java.io.IOException(s
   def this(s: String) = this(s, null)
 }
 
-@Singleton
-class WebSocketClientConfig @Inject() (val configuration: Config) {
+case class WebSocketClientConfig(maxFrameLength: Int)
```

Comment:
This is not necessarily a definitive solution.
While implementing, I also realized that there was a difference (a flaw?) with the `ReadSideConfig` which is that the root key ("lagom.persistence.read-side") is not specified by the constructor but by the caller. Maybe it is intended to have potentially different read-side config later (but even in this case, shouldn't them have the same root key) ?!? But anyway, somehow, this does not seem very consistent with the Kafka, or `CircuitBreakerConfig` which encapsulate their root config key.

Anyway, here I used the `case class` approach but I also integrated the root config key within the constructor. This is just a proposal and now I am thinking that it could actually be cleaner to use a `sealed trait` solution to permit only constructions from config object.

Reply:
> While implementing, I also realized that there was a difference (a flaw?)

Hmmm, you're right. This looks inconsistent and it goes on: `ReadSideConfig` provides default value so it can be built using `ReadSideConfig()` or `ReadSideConfig(config)` while `KafkaConfig` requires a `config` instance in all cases. I think we're better off sticking to a `sealed trait` without default values so I'm inclined towards the pattern used in `KafkaConfig`.